### DISCUSSION TITLE
Use And operators instead of OR for the Search system

### DISCUSSION THREAD

**Discussion by Peter4661** [PrestaShop/PrestaShop#40185] [Ideas]
Attached modified file search.php which improves search results very specific. Tested on 1.7.5.1 
Suggest to replace current search file with Search_php_modidied.

See it in action : www.derks-wielersport.nl

[Search_PHP modified.zip](https://github.com/PrestaShop/PrestaShop/files/2938265/Search_PHP.modified.zip)

**Comment by khouloudbelguith** [comment]
Hi @Peter4661,

Could you please provide us a clear and concise description of what the problem is and  clear and concise description of any alternative solutions or features you've considered.

About your code suggestion, would you be willing to make a pull request on GitHub with your code suggestion?
https://github.com/PrestaShop/PrestaShop/tree/develop
Thank you!

**Comment by Peter4661** [comment]
Hi khouloudbelguith ,

Current  Prestashop search is based on Boolean searching 'OR' operator. 
My proposal is making use of search 'AND' operator.

OR
The OR operator retrieves all items which contain either or both of the terms it separates. This retrieves more hits than searching one of the terms on its own.

For example in our shop ( chain 11 speed Shimano)
If you wish to search for the words "chain" or "11" or "speed" or "Shimano" in a phrase. For example: "chain or 11 or speed or shimano" will find the whole phrase 'chain or 11 or speed or shimano'. Delivering non specific search results.


AND
The AND operator retrieves all items which contain both of the terms it separates. This normally retrieves fewer hits than searching one of the terms on its own.

For example:

If you wish to search for the words "Chain" and "11" and "speed" and "Shimano in a phrase. AND operator will find the whole phrase containing 'Chain and 11 and speed and Shimano'. Delivering more precies search results.


I will make pull request on Github

**Comment by khouloudbelguith** [comment]
@Peter4661, thanks for these clarifications.
@marionf, @colinegin what do you think of his suggestion?
Thanks!

**Comment by colinegin** [comment]
@eternoendless @PrestaShop/prestashop-core-developers what do you think ?

**Comment by PierreRambaud** [comment]
@marionf @LouiseBonnard The search system has been redo with the levenshtein algorithm.
Is it still relevant?

**Comment by LouiseBonnard** [comment]
It looks complementary since the Levenshtein algorithm is about typing approximate words to catch the proper words. It might be a search parameter: enable either the _and_ or the _or_ operator in the store's search system.
